Jenkinsfile

```diff
@@ -11,109 +11,192 @@ pipeline {
     environment {
         DOCS = 'distribution/docs'
         ITESTS = 'distribution/test/itests/test-itests-alliance'
+        POMFIX = 'libs/libs-pomfix,libs/libs-pomfix-run'
+        LARGE_MVN_OPTS = '-Xmx8192M -Xss128M -XX:+CMSClassUnloadingEnabled -XX:+UseConcMarkSweepGC '
+        LINUX_MVN_RANDOM = '-Djava.security.egd=file:/dev/./urandom'
     }
     stages {
         stage('Setup') {
             steps {
                 slackSend color: 'good', message: "STARTED: ${JOB_NAME} ${BUILD_NUMBER} ${BUILD_URL}"
             }
         }
-        stage('Parallel Build') {
+        // Use the pomfix tool to validate that bundle dependencies are properly declared
+        stage('Validate Poms') {
+            agent { label 'linux-small' }
+            steps {
+                retry(3) {
+                    checkout scm
+                }
+                withMaven(maven: 'M3', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LINUX_MVN_RANDOM}') {
+                    sh 'mvn clean install -DskipStatic=true -DskipTests=true -pl $POMFIX'
+                }
+            }
+        }
+        // The incremental build will be triggered only for PRs. It will build the differences between the PR and the target branch
+        stage('Incremental Build') {
```

Comment:
We may need to consider removing the CRON statement -- have you been able to validate how the incremental build stage behaves on a CRON triggered build (nightly)? Does it just skip this step, or will it cause a build failure? I am also curious whether 'build now' will still work.

Reply:
As soon this is merged, will it cause Jenkins to flood with a build for every open PR?

Reply:
I believe it is possible to have the cron statement be valid only for the master branch if that would help